src/core/session.cpp

```diff
@@ -134,6 +134,57 @@ session::session(connection_pool & pool)
     backEnd_ = pooledSession.get_backend();
 }
 
+session::session(session && other)
+    : query_stream_(std::move(other.query_stream_)),
+      query_transformation_(std::move(other.query_transformation_)),
+      logger_(std::move(other.logger_)),
+      lastConnectParameters_(std::move(other.lastConnectParameters_)),
+      uppercaseColumnNames_(std::move(other.uppercaseColumnNames_)),
+      backEnd_(std::move(other.backEnd_)),
+      gotData_(std::move(other.gotData_)),
+      isFromPool_(std::move(other.isFromPool_)),
+      poolPosition_(std::move(other.poolPosition_)),
+      pool_(std::move(other.pool_))
+{
+    other.isFromPool_ = false;
+    other.pool_ = nullptr;
+    other.backEnd_ = nullptr;
+    other.pool_ = nullptr;
```

Comment:
I think it would make sense to add `reset_after_move()` or `clear_after_move()` private helper and call it from both here and move assignment operator.

Reply:
We could use placement new to re-initialize the fields like below.

`
    new (&other) session ();
`

Perhaps this is not a bad thing. It just made me realize that the 'once' and 'prepare' members get goofed up in the newly move constructed session.

Similarly, invoking the dtor explicitly can reduce some code duplication (if that's your concern):

`
    this->~session ();
`

But I didn't know where you stand with those two ways of doing things. Let me know which one you prefer.

Reply:
I'd strongly prefer a clearly named member function rather than using tricks with placement new or explicit dtor call.